cpp/src/arrow/extension/fixed_shape_tensor_test.cc

```diff
@@ -462,4 +543,121 @@ TEST_F(TestExtensionType, ToString) {
   ASSERT_EQ(expected_3, result_3);
 }
 
+TEST_F(TestExtensionType, GetScalar) {
```

Comment:
Is this test exercising anything specific to fixed_shape_tensor? `GetScalar` for extension types presumably already works, not sure this is worth testing again here.